### ISSUE TITLE
BUG - Firing bullets nudges player

### ISSUE THREAD

**Issue by mfreidel** [FreeDelete-Software/learn2shmup#3]
The initial implementation of firing bullets causes the bullet to "nudge" the player in various directions. 

### Steps To Reproduce

1. Run main scene (level0) from editor
2. Fire bullets
3. Player is moved upon firing

### Expected behavior

The player's position should not be altered when firing and/or the player object should not interact with its own bullets.

### Test environment details

Describe the environment you found this bug in using the following list:

- **Game Version:** 0.1.1
- **Commit Hash (if known):** https://github.com/FreeDelete-Software/learn2shmup/commit/4438fe989b0112b119bb45da45084d898cc72ea9
- [X] This problem occurs in the Godot editor

### Submission Checklist

Check all that apply:

- [X] Bug exists in the `devel` branch
- [X] This bug has not yet been reported.
- [X] This has not been fixed in newer game versions (including the `devel` branch)
- [X] The testing environment used an unmodified version of the codebase from the `FreeDelete-Software/learn2shmup` repo
- [x] I have tested this to the best of my ability
- [x] Further testing is necessary for a proper fix
- [x] I know how to how to fix this bug
- [X] I have read, selected, **and verified** the above checkmarks accurately

### Additional context

I introduced this early on and never got around to fixing it.

**Comment by mfreidel** [comment]
If I'm being honest, I didn't actually test this the best I could (removed checkmark). 

However, I have an unconfirmed hypothesis that this is caused by the bullet instance being created with the exact same position as the player. Simply adding an offset to the x axis when the instance is created could _possibly_ resolve the issue. I'll test this eventually.

**Comment by mfreidel** [comment]
Did some actual testing with the code. The offset for the projectile must be 32 (the length of the sprite), which fixes the documented bug.

~However, I've discovered an additional bug where you cannot fire while moving up and left. Not sure why this is, even after some playing around. Something worth noting here is that when moving up and left, the player object is moving in a negative direction on both axes. Maybe we need to see if it is an issue with the way movement has been implemented.~
I was having a keyboard ghosting issue while working on this and resolved it by using a better keyboard. Considering that potato-class hardware is our target environment, we may want to assume that keyboard ghosting is likely to occur.

As a side note, I found out that the `CollisionShape2D` node for the player object can be changed to the `CollisionPolygon2D` type in order to create a collision shape which more closely matches the artwork.
